Choice of the default environment

flexiti opened the pull request:
You can choose the default environment here

flexiti committed:
Merge pull request #2 from letscontrolit/mega

New wersion update

flexiti committed:
Merge pull request #3 from letscontrolit/mega

ver 19.04.2018

flexiti committed:
Merge pull request #4 from letscontrolit/mega

ver 20.04.2018

flexiti committed:
Merge pull request #5 from letscontrolit/mega

ver 21042018

flexiti committed:
Merge pull request #6 from letscontrolit/mega

v_22042018

flexiti committed:
Merge pull request #7 from letscontrolit/mega

update 20180424

flexiti committed:
default set

flexiti committed:
Update platformio.ini

flexiti committed:
Merge pull request #8 from letscontrolit/mega

Version 20180424

flexiti committed:
Merge pull request #9 from letscontrolit/mega

v30280424_2

flexiti committed:
Merge pull request #10 from letscontrolit/mega

ver 20180425

flexiti committed:
Platformio ini changes

flexiti committed:
Merge branch 'mega' of https://github.com/flexiti/ESPEasy into mega

TD-er commented:
Is it also possible to add some pre-select filtering to the sets?
I switch a lot between 'normal 4096' and 'esp32' and then click build/upload and sometimes clean.

flexiti commented:
I am using this because shortcuts on top ("Build", "Download" etc.)  just act 
or just "Run other target.." but for mi this is more complicated ;-)
What do you mean  

> add some pre-select filtering to the sets

 ?

TD-er commented:
If you open the build select dialog in Atom/PlatformIO. You get a long list.
In this dialog I start typing 'normal 4096' to get a smaller subselection.
Maybe there is also some other option in PlatformIO, that does allow to make a preselection of frequently used items that are always on top of that list.

Simply merging this PR will make the default for anyone set to ESP32. So that is not the best solution I guess.

flexiti commented:
For me, it's a simple solution, as the first time I launched this project it started to compile me into infinity (many versions) and I was a little lost;)
I assume that a large majority needs only one choice, you can set env:normal_ESP8266_1024 etc as default ... but this is only my opinion 😄 The decision is yours 😃

TD-er commented:
The 1024 versions are mainly for the Sonoff and ESP01 modules.
A lot of users use NodeMCU or Wemos D1 mini and these have 4 MB flash.

But I agree there are way too many options and I hope to be able to reduce them or at least when adding new ones their purpose is clear immediately (like Sonoff POW)

Oxyandy commented:
In platformIO, I press F7, then type 1024 - I then select the single build I want..
Another firmware I build has an excellent selection from pressing F7, seems clearer & more friendly than ESPeasy has, when I see it again I'll post back because is an excellent example

flexiti commented:
This change does not exclude this possibility

flexiti committed:
Merge pull request #11 from letscontrolit/mega

Change use of core 2_4_0  to 2_3_0

flexiti committed:
Merge pull request #12 from letscontrolit/mega

20180428 update

flexiti committed:
rulesTimers accuracy

So far, because rulesTimers() is checked asynchronously to timePassedSince time could be longer by 1 second from the set one
(especially important in short times)
This workaround decreases error to 100 ms.

flexiti committed:
Possibility to set the startup environment

TD-er commented on src/Misc.ino:
```diff
@@ -2629,7 +2629,7 @@ void rulesTimers()
     if (!RulesTimer[x].paused && RulesTimer[x].timestamp != 0L) // timer active?
     {
-      if (timeOutReached(RulesTimer[x].timestamp)) // timer finished?
+      if (timeOutReached(RulesTimer[x].timestamp - 100)) // timer finished?  (minus 100 ms)
```
You should not subtract from a time.
The time will wrap (unsigned int), but subtracting from that will result in strange results.
This will be after either 47.9 days, or half that period.

You could use `timediff`, which returns a signed int, which can be compared.

flexiti replied:
This will be ok ?

   -- if (timeOutReached(RulesTimer[x].timestamp - 100)) // timer finished?  (minus 100 ms)

  ++ if (timeDiff(millis(),RulesTimer[x].timestamp)  < 100)

Ps. I decided for 100 ms because  runOncePerSecond()  is not quite quite accurate and repeatable
as it should

TD-er replied:
I always have to check in the code to see the definition of timeDiff, so perhaps there should be a function called `timeUntil`.

But that's the general idea indeed.
And if rules timers have to be more accurate, we should add something to do just that, instead of applying some average correction offset. Maybe the suggested cron timer approach suggested in another issue would be a good addition for this.
Or we should add some timer to be set in the rules at which time the rules should be processed.
I want to fix it in such a way that it is beneficial for more use cases.
And applying a fixed offset to correct some timing issue is almost always not the best solution.

flexiti replied:
Globally, this is not important, but when I set the timer for 1 second I had 2 seconds, when I set it for 2 seconds, I had 3 seconds (100% and 50% error). With longer times the error is acceptable

flexiti replied:
ok, I close this pull
